### ISSUE TITLE
Unsupervised Representation Learning with Deep Convolutional Generative Adversarial Networks

### ISSUE THREAD

**Issue by omoindrot** [omoindrot/papers#6]
https://arxiv.org/abs/1511.06434
Alec Radford, Luke Metz ([indico.io]())
Soumith Chintala (FAIR)
### What are Generative Adversarial Networks?

Introduced by Ian GoodFellow (formerly at Google, now at OpenAI) in 2014: https://arxiv.org/abs/1406.2661

> We simultaneously train two models: a generative model G that captures the data distribution, and a discriminative model D that estimates the probability that a sample came from the training data rather than G. The training procedure for G is to maximize the probability of D making a mistake. This framework corresponds to a minimax two-player game
### Contributions of the paper
- make GAN more stable by introducing **Deep Convolutional Generative Adversarial Networks** (DCGANs)
- use the trained discriminator for image classification
- the generators have vector arithmetic properties
## New model architecture: DCGANs

![generator](https://cloud.githubusercontent.com/assets/13662086/17780878/4968485a-656d-11e6-9576-2d073c8f6405.png)

Using new insights to improve the stability of GANs (there are known to be hard to train).

**Key elements:**
- use an all convolutional net for the generator and discriminator
  - max pooling is replaced by strided convolutions
  - upsampling in the generator is done with **fractionally-strided convolutions**
- eliminate fully connected layers
  - use global average pooling at the end
  - only 1 fully connected layer
- use **batch normalization**
  - stabilizes learning
  - no batch norm in first layer of discriminator and last layer of generator
- use ReLU in generator
  - except output layer with tanh
- use LeakyReLU in the discriminator
